aws-rds-dbinstance/src/main/java/software/amazon/rds/dbinstance/CreateHandler.java

```diff
@@ -0,0 +1,98 @@
+package software.amazon.rds.dbinstance;
+
+import software.amazon.awssdk.services.rds.RdsClient;
+import software.amazon.awssdk.utils.StringUtils;
+import software.amazon.cloudformation.proxy.AmazonWebServicesClientProxy;
+import software.amazon.cloudformation.proxy.Logger;
+import software.amazon.cloudformation.proxy.ProgressEvent;
+import software.amazon.cloudformation.proxy.ProxyClient;
+import software.amazon.cloudformation.proxy.ResourceHandlerRequest;
+import software.amazon.cloudformation.resource.IdentifierUtils;
+
+
+public class CreateHandler extends BaseHandlerStd {
+    protected ProgressEvent<ResourceModel, CallbackContext> handleRequest(
+        final AmazonWebServicesClientProxy proxy,
+        final ResourceHandlerRequest<ResourceModel> request,
+        final CallbackContext callbackContext,
+        final ProxyClient<RdsClient> proxyClient,
+        final Logger logger) {
+
+        final ResourceModel model = request.getDesiredResourceState();
+
+        if (StringUtils.isEmpty(model.getDBInstanceIdentifier())) {
```

Comment:
TODO: move this to ProgressEvent.progress().then(), otherwise a new name would be generated every time create request fails, which is fairly dangerous.